language/diem-framework/modules/0L/Oracle.move

```diff
@@ -266,6 +267,26 @@ address 0x1 {
           total_weight: 0,
         };
       }
+
+      /// This function will clear ballots if the current voting round has expired 
+      /// Unlike enter_new_upgrade_round, it will not start a new round 
+      /// it also does not clear the consensus so that it can be checked by the vm later
+      /// (that only happens if someone submits a proposal)
+      public fun clear_expired_ballots(vm: &signer) acquires Oracles {
+        Roles::assert_diem_root(vm);
+        let current_height = BlockHeight::get_height();
+        let upgrade_oracle = &mut borrow_global_mut<Oracles>(CoreAddresses::DIEM_ROOT_ADDRESS()).upgrade;
+  
+        // check if qualifies as a new round
+        let is_new_round = current_height > upgrade_oracle.vote_window;
+
+        if (is_new_round) {
+          upgrade_oracle.validators_voted = Vector::empty<address>();
```

Comment:
I think we should move line 277 `let upgrade_oracle = &mut borrow_global_mut<Oracles>(CoreAddresses::DIEM_ROOT_ADDRESS()).upgrade;` here